### PR TITLE
ROC-5448 Copy metadata contexts for custom validators

### PR DESCRIPTION
**Pull request by andrewwa-kainos**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-5448

### Change description ###
https://github.com/typestack/class-validator/pull/296 addressing https://github.com/typestack/class-validator/issues/292

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by robertparkinson — APPROVED** [review]

File: test/functional/validation-options.spec.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import "es6-shim";
-import {Contains, Matches, MinLength, ValidateNested} from "../../src/decorator/decorators";
+import {Contains, Matches, MinLength, ValidateNested, IsDefined} from "../../src/decorator/decorators";
```
Comment (sabahirfan) [inline test/functional/validation-options.spec.ts:2]: would be nice to remove relative paths in imports here

**Review by sabahirfan — APPROVED** [review]

File: test/functional/validation-options.spec.ts (hunk 1 above)
Comment (andrewwa-kainos) [reply test/functional/validation-options.spec.ts:2]: As it's a forked project, we're trying to make as few changes as possible.

Comment (sabahirfan) [reply test/functional/validation-options.spec.ts:2]: 👍

**Comment by gh-clement-taboulot** [comment]
@andrewwa-kainos is this will be merged on typestack/class-validator ?

**Comment by andrewwa-kainos** [comment]
@taboulot 
> andrewwa-kainos is this will be merged on typestack/class-validator ?

Unknown... the ticket there is still open and unassigned

**Comment by prakashchokalingam** [comment]
Waiting for the typestack/class-validator merge  ☹️

**Pull request merged**